agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt

```diff
@@ -51,6 +50,55 @@ class OfflineAgendaRepository(
         }.launchIn(scope)
     }
 
+    override fun updateAgendaItem(agendaItem: AgendaItem) {
+        when(agendaItem){
+            is AgendaItem.Event -> {
+                networkApiCall {
+                    val eventBody = agendaItem.toEventBody()
+                    val eventBodyPart = buildEventBodyPart(eventBody)
+                    val photosPart = buildPhotosPart(agendaItem.photos)
+                    taskyRetrofitApi.updateEvent(
+                        eventBodyPart = eventBodyPart,
+                        photosPart = photosPart
+                    )
+                }
+
+            }
+            is AgendaItem.Reminder -> {
+                networkApiCall {
+                    taskyRetrofitApi.updateReminder(agendaItem.toReminderBody())
+                }
+            }
+            is AgendaItem.Task -> {
+                networkApiCall {
+                    taskyRetrofitApi.updateTask(agendaItem.toTaskBody())
+                }
+            }
+        }.launchIn(scope)
+    }
+
+    override fun deleteAgendaItem(agendaItem: AgendaItem) {
+        when(agendaItem){
+            is AgendaItem.Event -> {
+                networkApiCall {
+                    taskyRetrofitApi.deleteEvent(agendaItem.id)
+                }
+            }
```

Comment:
Also, here you delete the agenda item on the remote server, but your app will still have it locally